### ISSUE TITLE
[ifit] not detecting QZ as a device

### ISSUE THREAD

**Issue by msuvat** [cagnulein/qdomyos-zwift#845]
Hi,

I'm on version 2.10.110 and have tried to connect iFit with QZ with a Pixel 6 and Galaxy S9, but iFit doesn't detect QZ as a device, even though QZ is already connected to my Renpho bike and displaying stats. Have tried both running iFit on both the P6 and S9 and QZ on the other and it's the same result.

Before going into logs, I've pasted below a screenshot of my QZ settings - does anything need to change here?

Thanks

![Screenshot_20220625-110133](https://user-images.githubusercontent.com/98742740/175752774-68eee17c-97ef-4c46-b368-2754c1515ed2.png)

**Comment by cagnulein** [comment]
Hi, on android you need to rename your device to I_TL and reboot it
Let me know
Il 25 giu 2022, 03:18 +0200, msuvat ***@***.***>, ha scritto:

**Comment by msuvat** [comment]
Thanks @cagnulein , it connects now! 

The auto resistance doesn't seem to work properly but will do more testing in the coming days, the ifit app itself is quite clunky.

Do I need to press play on the QZ app as well after I start the ifit workout or is that only for the purpose of recording workout stats which QZ emails to me afterwards?

**Comment by cagnulein** [comment]
Great!
You need to press start because otherwise I will not calculate wattage and
speed. So yes, press start :)
For auto resistance, the peloton conversion is scaled to ifit max and min,
so it should work as peloton for your bike (by the way, which bike do you
have?). Also remember that zwift resistance gain and offset affects the
auto resistance conversion.
Let me know
R.

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 25 giu 2022 alle ore 08:46 msuvat ***@***.***>
ha scritto:

**Comment by msuvat** [comment]
Thanks Roberto. I have the Renpho bike, auto resistance seems to be working ok now.

On another note, I can't seem to turn off the "heart rate service outside FTMS" setting, when I turn off, it comes back on again after I restart QZ, do you know why that may be the case?

The iFit app loses connection to QZ when it picks up my Bluetooth heart rate monitor directly, so trying to connect the HR monitor to QZ to relay to the iFit as a single FTMS signal. QZ is running on Pixel 6 with android 12. Ant+ cadence and heart are both off.

**Comment by cagnulein** [comment]
Hi, the heart rate outside ftms doesn't toggle on your phone because you are on android 10 or higher and so it's a limitation of your android. you can't broadcast the bike and your heart rate on different bluetooth services.
I don't remember if I handled the hr on the ifit protocol but it could be doable: in this way hr and bike metrics could be merged in one service
I will let you know
Il 26 giu 2022, 05:52 +0200, msuvat ***@***.***>, ha scritto:

**Comment by cagnulein** [comment]
No, I checked, no HR to iFit. Remember that ifit it's not using FTMS protocol, they have their own protocol (it's such a mess!) so you should ask them why when you connect an external HR they drops the connection to "the bike" (qz). It seems a bug in their app.

**Comment by msuvat** [comment]
Thanks for checking Roberto, much appreciated 🙂
